src/+otp/+campbellmooredae/CampbellMooreDAEProblem.m

```diff
@@ -0,0 +1,41 @@
+classdef CampbellMooreDAEProblem < otp.Problem
+    % Originally Index-3 DAE problem reduced to index 1
+    % Reference: Campbell, Stephen L., and Edward Moore.
+    % "Constraint preserving integrators for general nonlinear higher index DAEs."
+    % Numerische Mathematik 69.4 (1995): 383-399.
+    
+    methods
+        function obj = CampbellMooreDAEProblem(timeSpan, y0, parameters)
+            obj@otp.Problem('Campbell-Moore DAE', 10, timeSpan, y0, parameters);
+        end
+    end
+    
+    methods (Access = protected)
+        function onSettingsChanged(obj)
+            r = obj.Parameters.r;
+            rho = obj.Parameters.rho;
+            
+            obj.Rhs = otp.Rhs(@(t, y) otp.campbellmooredae.f(t, y, r,rho), ...
+                otp.Rhs.FieldNames.Jacobian, @(t, y) otp.campbellmooredae.jac(t, y, r,rho), ...
+                otp.Rhs.FieldNames.Mass,  otp.campbellmooredae.mass([], [], r,rho), ...
+                otp.Rhs.FieldNames.MStateDependence, 'none', ...
+                otp.Rhs.FieldNames.MassSingular, 'yes');
+        end
+        
+        function validateNewState(obj, newTimeSpan, newY0, newParameters)
+            validateNewState@otp.Problem(obj, newTimeSpan, newY0, newParameters)
+            
+            otp.utils.StructParser(newParameters) ...
+                .checkField('rho', 'scalar', 'real', 'finite', 'positive');
+            otp.utils.StructParser(newParameters) ...
+                .checkField('r', 'scalar', 'real', 'finite', 'positive');
```

Comment:
Reading the paper, I don't see a compelling reason for r and rho being positive, unless I am missing something major.

Reply:
oh I see it now. In fact it says that r > rho.

we need to add, "@(x) x > newParameters.rho" to the validation for r I think.

Reply:
another thing, we can more efficiently write this as
`otp.utils.StructParser(newParameters) ...
                .checkField('rho', 'scalar', 'real', 'finite', 'positive') ...
                .checkField('r', 'scalar', 'real', 'finite', 'positive', @(x) x > newParameters.rho);`